### PR TITLE
Run only listed tests on s390x

### PR DESCRIPTION
**Pull request by AlekseiNikiforovIBM**
Skip tests that are failing

This was previously part of https://github.com/pytorch/pytorch/pull/125401

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: test/ao/sparsity/test_composability.py [hunk 1]
```diff
@@ -75,6 +76,7 @@ class TestComposability(TestCase):
     # This test checks whether performing quantization prepare before sparse prepare
     # causes any issues and verifies that the correct observers are inserted and that
     # the quantized model works as expected
+    @unittest.skipIf(IS_S390X, "Currently fails on s390x")
```
Comment (huydhn) [inline test/ao/sparsity/test_composability.py:79]: Echo @malfet's point earlier, is this possible to just use `skipIfS390X` here (and all other places too)?


```suggestion
    @skipIfS390X
```

File: torch/testing/_internal/common_utils.py [hunk 2]
```diff
@@ -1842,6 +1843,21 @@ def wrap_fn(self, *args, **kwargs):
         return wrap_fn
     return dec_fn
 
+def skipIfS390X(func=None, *, msg="test doesn't currently work on the s390x stack"):
```
Comment (huydhn) [inline torch/testing/_internal/common_utils.py:1846]: This looks fine.  You could also consider having `xfailIfS390X`, as xfail the test is usually a better mechanism than skipping it

Comment (malfet) [reply torch/testing/_internal/common_utils.py:1846]: To elaborate a bit, `xfail` allows one to know if their change fixed something, while `skip` will leave the test in this state in perpetuity and likely cause author of the fix to write a new one, as they'll not be aware that skipped test existed

File: torch/testing/_internal/common_utils.py [hunk 3]
```diff
@@ -1842,6 +1843,24 @@ def wrap_fn(self, *args, **kwargs):
...
+        def wrapper(*args, **kwargs):
+            if platform.machine() == "s390x":  # noqa: F821
+                raise unittest.SkipTest(reason)
+            else:
+                return fn(*args, **kwargs)
+        return wrapper
+    if func:
+        return dec_fn(func)
+    return dec_fn
+
```
Comment (malfet) [inline torch/testing/_internal/common_utils.py:1860]: Not used anywhere, so let's deleted
```suggestion
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**